src/app/(root)/product/add/components/image-upload.tsx

```diff
@@ -0,0 +1,94 @@
+"use client"
+
+import { ChangeEvent, useState } from "react"
+import Image from "next/image"
+import { Button } from "@mijn-ui/components/button"
+import { Input } from "@mijn-ui/components/input"
+import { Label } from "@mijn-ui/components/label"
+import { FaX } from "react-icons/fa6"
+import { IconContext } from "react-icons/lib"
+import { LuImagePlus } from "react-icons/lu"
+
+export type ImageUploadField = {
+  id: string
+  name: string
+}
+
+const ImageUpload = ({ id, name }: ImageUploadField) => {
+  const [selectedImage, setSelectedImage] = useState<string>()
+
+  const handleImageUpload = (e: ChangeEvent<HTMLInputElement>) => {
+    const imageFile = e.target.files?.[0]
+
+    if (imageFile) {
+      const reader = new FileReader()
+      reader.onloadend = () => {
+        setSelectedImage(reader.result as string)
+      }
+      reader.readAsDataURL(imageFile)
+    }
+  }
+
+  const handleImageRemove = () => {
+    setSelectedImage("")
+  }
+
+  return (
+    <div className="flex flex-col gap-2">
+      <Label htmlFor="product-image">Product Image</Label>
+      <div className="relative w-max">
+        <div
+          className={`flex flex-col justify-center items-center gap-4 md:6 text-neutral-text rounded-md border border-main-border px-4 py-6 md:p-8 `}
+        >
+          <IconContext.Provider
+            value={{ className: "text-[3.5em] md:text-[5em]" }}
```

Comment:
Couldn't we achieve the same behaviour by doing this?

```diff
- <IconContext.Provider
-           value={{ className: "text-[3.5em] md:text-[5em]" }}
-         >
-           <LuImagePlus />
- </IconContext.Provider>

+ <LuImagePlus className: "text-[3.5em] md:text-[5em]" />
```

Let me know if using IconContext.Provider is necessary.